lib/search_counter.js

```diff
@@ -0,0 +1,156 @@
+'use strict'
+
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+
+
+let counter = {};
+
+module.exports = function (context){
+    const logger = context.foundation.makeLogger({module: 'teraserver_search_counter'});
+
```

Comment:
If there is no stats configuration provided the stats module needs to be disabled rather than trying to rely on defaults. In production environments there will be no `default` elasticsearch connection so this will break if deployed without the proper configuration.